src/stats/accumulated/mod.rs

```diff
@@ -7,6 +7,7 @@ use crate::protocol::Protocol;
 
 #[derive(Clone, Debug)]
 pub struct AccumulatedStatsInner {
+    pub last_message: Option<mavlink::MAVLinkV2MessageRaw>,
```

Comment:
```suggestion
    pub last_message: Option<Arc<Protocol>>,
```

This would be much, much less expensive

Reply:
If the message is not guaranteed to be released after use, make this a weak reference by downgrading the Arc instead